integration-tests/gradle/src/test/java/io/quarkus/gradle/BuildConfigurationTest.java

```diff
@@ -1,108 +1,142 @@
 package io.quarkus.gradle;
 
-import static org.assertj.core.api.Assertions.assertThat;
-
 import java.io.File;
 import java.io.IOException;
 import java.net.URISyntaxException;
 import java.nio.file.Path;
 
+import org.assertj.core.api.SoftAssertions;
+import org.assertj.core.api.junit.jupiter.InjectSoftAssertions;
+import org.assertj.core.api.junit.jupiter.SoftAssertionsExtension;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
 
+@ExtendWith(SoftAssertionsExtension.class)
 public class BuildConfigurationTest extends QuarkusGradleWrapperTestBase {
```

Comment:
Refactored the test to unify common test logic, to run less Gradle builds (faster test runtime) and also emit build output on failure.